test/over_react/component/error_boundary_test.dart

```diff
@@ -145,6 +167,446 @@ void main() {
+          errorSentToComponentDidCatchCallback = componentDidCatchCallbackArguments[0];
+          errorInfoSentToComponentDidCatchCallback = componentDidCatchCallbackArguments[1];
+        }
+
+        if (calls.length > 1) {
+          expect(calls[1].keys.single, 'onComponentIsUnrecoverable', reason: 'test setup sanity check');
+          final componentIsUnrecoverableCallbackArguments = calls[1]['onComponentIsUnrecoverable'];
+          if (componentIsUnrecoverableCallbackArguments != null) {
+            errorSentToComponentIsUnrecoverableCallback = componentIsUnrecoverableCallbackArguments[0];
+            errorInfoSentToComponentIsUnrecoverableCallback = componentIsUnrecoverableCallbackArguments[1];
+          }
+        }
+      }
+
+      setUp(() {
+        mountNode = new DivElement();
+        document.body.append(mountNode);
+
+        calls = [];
+
+        jacket = mount(
+          (ErrorBoundary()
+            ..identicalErrorFrequencyTolerance = const Duration(
+                milliseconds: identicalErrorFrequencyToleranceInMs)
+            ..onComponentDidCatch = (err, info) {
+              calls.add({'onComponentDidCatch': [err, info]});
+            }
+            ..onComponentIsUnrecoverable = (err, info) {
+              calls.add({'onComponentIsUnrecoverable': [err, info]});
+            }
+          )(
+            (Flawed()..addTestId('flawedComponent'))(
+              (Flawed()
+                ..buttonTestIdPrefix = 'nestedFlawedComponent_'
+                ..addTestId('nestedFlawedComponent')
+              )(),
+            ),
+          ),
+          mountNode: mountNode,
+        );
+
+        flawedRenderedInstance = getByTestId(jacket.getInstance(), 'flawedComponent');
+        expect(flawedRenderedInstance, isNotNull, reason: 'test setup sanity check');
+
+        nestedFlawedRenderedInstance = getByTestId(jacket.getInstance(), 'nestedFlawedComponent');
+        expect(nestedFlawedRenderedInstance, isNotNull, reason: 'test setup sanity check');
+      });
+
+      tearDown(() {
+        calls = null;
+        flawedRenderedInstance = null;
+        nestedFlawedRenderedInstance = null;
+        errorSentToComponentDidCatchCallback = null;
+        errorInfoSentToComponentDidCatchCallback = null;
+        errorSentToComponentIsUnrecoverableCallback = null;
+        errorInfoSentToComponentIsUnrecoverableCallback = null;
+        mountNode.remove();
+        mountNode = null;
+      });
+
+      group('and consecutive errors are thrown from the same component', () {
+        Future<Null> triggerErrorsThatSignifyAnUnrecoverableComponent() async {
+          getFlawedButtonNode().click();
+          expect(calls.length, 1, reason: 'test setup sanity check');
+          expect(calls.single.keys.single, isNot('onComponentIsUnrecoverable'), reason: 'test setup sanity check');
+
+          calls.clear();
+          await new Future.delayed(const Duration(milliseconds: identicalErrorFrequencyToleranceInMs - 50));
```

Comment:
#nit magic number `50`.

In tests like these, I like to just do multiples of the main number to avoid having to keep all these in sync:
```suggestion
          await new Future.delayed(identicalErrorFrequencyTolerance ~/ 2);
```

Reply:
Addressed in https://github.com/Workiva/over_react/pull/350/commits/a9e864eeef2abd045c977aa1377569d7fcb3d399